File: ProjectEuler/Problem_7/7-10001st_prime.cs
// What is the 10001st prime number?

void Main()
{
    int n = 1;
    double i = 3;

    while (n != 10000)
    {
        if (isPrime(i))
        {
            n += 1;
        }
        i += 2;
    }
    Console.WriteLine(i - 2);
}

// Define other methods and classes here
bool isPrime(double num)
{
    if (num < 2) return false;
    if (num == 2 || num == 3) return true;
    if (num % 2 == 0 || num % 3 == 0) return false;
    foreach (int i in Enumerable.Range(3, (int)Math.Sqrt(num) + 1))
    {
        if (num % i == 0) return false;
    }
    return true;
}
// 10001 prime is 104743

File: ProjectEuler/Problem_7/7-10001st_prime.js
// What is the 10001st prime number?

const is_prime = x => {
    if (x < 2) return false
    if (x == 2 || x == 3 || x == 5) return true
    if (x % 2 == 0 || x % 3 == 0 || x % 5 == 0) return false
    for (let i = 3; i < Math.sqrt(x) + 1; i += 2){
        if (x % i == 0) return false
    }
    return true
}

let i = 2
let n = 0

while (n != 10001) {
    if (is_prime(i)) n += 1
    i += 1
}

console.log(i-1)  // 104743